staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder/builder.go

```diff
@@ -123,44 +130,44 @@ func BuildSwagger(crd *apiextensions.CustomResourceDefinition, version string, o
 	routes := make([]*restful.RouteBuilder, 0)
 	root := fmt.Sprintf("/apis/%s/%s/%s", b.group, b.version, b.plural)
 	if b.namespaced && b.tenanted {
-		routes = append(routes, b.buildRoute(root, "", "GET", "list", sampleList).
+		routes = append(routes, b.buildRoute(root, "", "GET", "list", "list", sampleList).
 			Operation("list"+b.kind+"ForAllNamespacesAllTenants"))
 
 		root = fmt.Sprintf("/apis/%s/%s/tenants/{tenant}/%s", b.group, b.version, b.plural)
-		routes = append(routes, b.buildRoute(root, "", "GET", "list", sampleList).
-			Operation("list"+b.kind+"ForAllNamespacesUnderOneTenatn"))
+		routes = append(routes, b.buildRoute(root, "", "GET", "list", "list", sampleList).
+			Operation("list"+b.kind+"ForAllNamespacesUnderOneTenant"))
 
 		root = fmt.Sprintf("/apis/%s/%s/tenants/{tenant}/namespaces/{namespace}/%s", b.group, b.version, b.plural)
 	}
 
 	if !b.namespaced && b.tenanted {
-		routes = append(routes, b.buildRoute(root, "", "GET", "list", sampleList).
+		routes = append(routes, b.buildRoute(root, "", "GET", "list", "list", sampleList).
 			Operation("list"+b.kind+"ForAllTenants"))
 		root = fmt.Sprintf("/apis/%s/%s/tenants/{tenant}/%s", b.group, b.version, b.plural)
 	}
 
-	routes = append(routes, b.buildRoute(root, "", "GET", "list", sampleList))
-	routes = append(routes, b.buildRoute(root, "", "POST", "create", sample).Reads(sample))
-	routes = append(routes, b.buildRoute(root, "", "DELETE", "deletecollection", status))
+	routes = append(routes, b.buildRoute(root, "", "GET", "list", "list", sampleList))
```

Comment:
are there any reasons that calling buildRoute(...) with one more arg than previous?

Reply:
We cherry-picked testcase "TestCRDRouteParameterBuilder". Multi tenancy support and also community PR: https://github.com/kubernetes/kubernetes/pull/80074 are necessary to get this test case pass.

Reply:
thanks for the explanation. got it.